### PR TITLE
Feat/pci block storages

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | DTCORE-1762
| License          | BSD 3-Clause

## Description

Migrate pci block storage section to micro app

### CODE REVIEW

**Review by kqesar — DISMISSED** [review]

**Review by cbourgois — DISMISSED** [review]

**Review by cbourgois — APPROVED** [review]

**Review by kqesar — APPROVED** [review]

**Review by kqesar — DISMISSED** [review]

**Review by cbourgois — DISMISSED** [review]

File: packages/manager/modules/pci/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "@ovh-ux/manager-filters": "^1.1.1",
     "@ovh-ux/manager-models": "^1.0.0",
     "@ovh-ux/manager-ng-layout-helpers": "^2.0.0",
+    "@ovh-ux/manager-pci-block-storage": "^0.0.0",
```
Comment (qpavy) [inline packages/manager/modules/pci/package.json:43]: not needed as a micro app is independent of other apps

File: packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js [hunk 2]
```diff
@@ -4,9 +4,7 @@ import '@uirouter/angularjs';
 import 'angular-translate';
 import '@ovh-ux/ui-kit';
 import 'ovh-api-services';
-
-import blockStorages from '../../../blocks';
-import volumeEdit from '../../../blocks/block/edit/volume-edit';
+import pciBlockStorage from '@ovh-ux/manager-pci-block-storage';
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js:7]: not needed as a micro app is independent of other apps

File: packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js [hunk 3]
```diff
@@ -19,8 +17,7 @@ angular
     'ovh-api-services',
     'ngTranslateAsyncLoader',
     'pascalprecht.translate',
-    blockStorages,
-    volumeEdit,
+    pciBlockStorage,
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js:20]: not needed as a micro app is independent of other apps

File: packages/manager/modules/pci/src/projects/project/storages/storages.module.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import angular from 'angular';
 import '@ovh-ux/ui-kit';
 import '@ovh-ux/manager-filters';
 
-import blocks from './blocks';
+import pciBlockStorage from '@ovh-ux/manager-pci-block-storage';
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/storages/storages.module.js:5]: not needed as a micro app is independent of other apps

File: packages/manager/modules/pci/src/projects/project/storages/storages.module.js [hunk 5]
```diff
@@ -23,11 +23,11 @@ angular
     'ui.router',
     'ovhManagerFilters',
     storagesComponents,
-    blocks,
     cloudArchive,
     databases,
     instanceBackups,
     objectStorage,
+    pciBlockStorage,
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/storages/storages.module.js:30]: not needed as a micro app is independent of other apps

Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/storages.module.js:30]: it's imported to allow feature flipping between uapp & monolith ; once the uapp is ok we will remove the imported module from pci

File: packages/manager/modules/pci/src/projects/project/storages/storages.module.js (hunk 4 above)
Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/storages.module.js:5]: see https://github.com/ovh/manager/pull/10563#discussion_r1472797974

File: packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js (hunk 3 above)
Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js:20]: https://github.com/ovh/manager/pull/10563#discussion_r1472797974

File: packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js (hunk 2 above)
Comment (frenautvh) [reply packages/manager/modules/pci/src/projects/project/storages/snapshots/snapshot/create-volume/create-volume.module.js:7]: https://github.com/ovh/manager/pull/10563#discussion_r1472797974

File: packages/manager/modules/pci/package.json (hunk 1 above)
Comment (frenautvh) [reply packages/manager/modules/pci/package.json:43]: https://github.com/ovh/manager/pull/10563#discussion_r1472797974

File: packages/manager/apps/pci-block-storage/src/pci-block-storage.routing.js
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-block-storage/src/pci-block-storage.routing.js:1]: the routing is needed for the application to be working both inside the container and outside of it ; we tested both cases. Did i miss something ?

File: packages/manager/apps/pci-block-storage/src/index.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    setShellClient(client);
+    return client.environment.getEnvironment();
+  })
+  .then((environment) =>
+    import(`./config-${environment.getRegion()}`)
+      .catch(() => {})
+      .then(() => import('./pci-block-storage.module'))
+      .then(({ default: startApplication }) =>
+        startApplication(document.body, getShellClient()),
+      ),
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-block-storage/src/index.js:26]: see my other comment we are using the getShellClient in other part of the code to retrieve the shell client api in order to generate some URLs

**Pull request closed without merging**